src/main/java/io/dockstore/provision/GSPlugin.java

```diff
@@ -0,0 +1,271 @@
+                    for (Map.Entry<String, String> entry : map.entrySet()) {
+                        System.out.println("Loading " + entry.getKey() + "->" + entry.getValue());
+                    }
+                    blobInfo = BlobInfo.newBuilder(blobId).setContentType(contentType).setMetadata(map).build();
+                } catch (com.google.gson.JsonSyntaxException ex) {
+                    System.err.println(
+                            "Could not load metadata. Metadata syntax exception. Uploading file without metadata:" + ex.getMessage());
+                    blobInfo = BlobInfo.newBuilder(blobId).setContentType(contentType).build();
+                }
+            } else {
+                blobInfo = BlobInfo.newBuilder(blobId).setContentType(contentType).build();
+            }
+
+            ProgressPrinter printer = new ProgressPrinter();
+
+            if (inputSize > MAX_FILE_SIZE_FOR_SINGLE_WRITE) {
+                // When content is not available or large (1MB or more) it is recommended
+                // to write it in chunks via the blob's channel writer.
+                try {
+                    WriteChannel writer = gsClient.writer(blobInfo);
+                    byte[] buffer = new byte[UPLOAD_BUFFER_SIZE];
+                    try (InputStream input = Files.newInputStream(sourceFile)) {
+                        int limit;
+                        long runningTotal = 0;
+                        while ((limit = input.read(buffer)) >= 0) {
+                            try {
+                                writer.write(ByteBuffer.wrap(buffer, 0, limit));
+                                runningTotal = runningTotal + limit;
+                                printer.handleProgress(runningTotal, inputSize);
+                            } catch (Exception ex) {
+                                System.err.println("Could not write file:" + ex.getMessage());
+                                ex.printStackTrace();
+                                return false;
+                            }
+                        }
+                    } catch (FileNotFoundException e) {
+                        System.err.println("File not found exception:" + e.getMessage());
+                        return false;
+                    }
+                    writer.close();
+                } catch (IOException e) {
+                    System.err.println("Could not upload file. IO exception:" + e.getMessage());
+                    return false;
+                }
+            } else {
+                byte[] bytes;
+                try {
+                    bytes = Files.readAllBytes(sourceFile);
+                } catch (IOException e) {
+                    System.err.println("Could not read file to upload. IO exception:" + e.getMessage());
+                    return false;
+                }
+                // create the blob in one request.
+                gsClient.create(blobInfo, bytes);
```

Comment:
What happens when this fails? Does it throw an exception or do you need to check the return value? Seems like you need to handle that case.

Reply:
try catch added